### PR TITLE
year 0 and ISO 8601 and slider

### PR DESCRIPTION
**Pull request by gregallensworth**
Refer to https://github.com/OpenHistoricalMap/issues/issues/624

I have made the changes on a branch, but would like them tested before I merge into main, since OHM loads directly from the main branch on Github.

Testing is as follows:

- check out the code `git clone https://github.com/OpenHistoricalMap/leaflet-ohm-timeslider-v2.git`
- switch to the new ISO-8601 branch: `git checkout bce-iso8601-compliance`
- start a Python web server: `python -m SimpleHTTPServer 9646` or `python3 -m http.server 9646`
- point browser at http://localhost:9646/ and try it out

### CODE REVIEW

File: leaflet-ohm-timeslider.js [hunk 1]
```diff
@@ -253,6 +253,12 @@ L.Control.OHMTimeSlider = L.Control.extend({
             this.setDateRangeFormAsOutOfSync(true);
         });
         L.DomEvent.on(this.controls.slider, 'input', () => {
+            // hack for <0 and >= -1 so we never have a -0.xxx value; without this "0" (1 BCE) exists twice whilst sliding
+            // this does make a "funny step" in the range input between -0.999999 and -0.000001 but that beats seeing 1 BCE twice
```
Comment (1ec5) [inline leaflet-ohm-timeslider.js:257]: −0.5 is a valid value (around July of 1&nbsp;BCE). I can get to that date by entering `-0000-07-01` into the Change Date dialog but not by dragging the slider.



Bringing up the Change Date dialog again formats the date as “7/1/0”, which is not a valid formatted date. It should say either “-0000-07-01” (raw ISO date) or “7/1/-1” (formatted MDY date). This is a single year being represented as three different numbers simultaneously: “-1”, “0”, and “1 (BCE)”.



If we can’t reuse the dropdown menus in the Change Date dialog, I suggest changing it to not format the date and only accept a raw ISO date for consistency with the range control’s year fields.

Comment (gregallensworth) [reply leaflet-ohm-timeslider.js:257]: > −0.5 is a valid value (around July of 1 BCE)

It is valid, but is the same output as +0.5 If I do not have that workaround, then the slider effectively has two copies of 1 BE, the -1 to 0 range and the 0 to 1 range.


> I can get to that date by entering -0000-07-01 into the Change Date dialog but not by dragging the slider.

It is working for me. See screengrab below. Can you describe the problem in more detail?

timeslider scroll through 1 BCE

Comment (1ec5) [reply leaflet-ohm-timeslider.js:257]: > It is valid, but is the same output as +0.5 If I do not have that workaround, then the slider effectively has two copies of 1 BE, the -1 to 0 range and the 0 to 1 range.

I made a typo above yet again. To clarify, this is what I would expect:

Slider `value` | Change Date in ISO format | Filter `decdate` | Formatted display
---:|----|---:|----
−1 | `-0001-01-01` | −1 | 2 BCE
−0.5 | `-0001-07-01` | −0.5 | 2 BCE
−0 | `-0000-01-01` | −0 | 2 BCE (?)
0 | `0000-01-01` | 0 | 1 BCE
0.5 | `0000-07-01` | 0.5 | 1 BCE
1  | `0001-01-01`| 1 | 1

> It is working for me. See screengrab below. Can you describe the problem in more detail?

https://github.com/OpenHistoricalMap/leaflet-ohm-timeslider-v2/assets/1231218/d637d584-6c1e-44f2-baaf-3d963a3c21b9

**Comment by gregallensworth** [comment]
I'm going to close this PR and the bce-iso8601-compliance branch, and revisit from a different angle, fixing the decimaldate math. I think Minh is right that the best order of operations is:
- localization stuff in https://github.com/OpenHistoricalMap/issues/issues/626 which can result in a few simplifications and dependencies
- fix `decimaldate.decimal2iso()` so that -0.5 and +0.5 are not the same (-1 from year if <0)
- then UI improvements to prevent confusion as to why 0 becomes -1, =1 becomes -2, etc.

**Pull request closed without merging**